url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/HttpHistory.java

```diff
@@ -0,0 +1,59 @@
+package net.hetimatan.net.http;
+
+import java.io.File;
+import java.io.IOException;
+
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.event.GlobalAccessProperty;
+import net.hetimatan.util.log.Log;
+
+
+public class HttpHistory {
+
+	private CashKyoroFile mCash = null;
+	private static HttpHistory sHistory = null;
+
+	private HttpHistory() throws IOException {
+		File parent = (new File("dummy")).getAbsoluteFile().getParentFile();
+		String path = GlobalAccessProperty.getInstance().get("my.home", parent.getAbsolutePath());
+		File home = new File(path);
+		mCash = new CashKyoroFile(new File(home, "http.history"), 512, 2);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *mCash*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).